### PR TITLE
Add Io code for LTC6813 Cell Temperatures

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

- Added Io code for LTC6813 GPIO voltage measurement
- Converted GPIO voltages to the corresponding cell temperatures

### Testing Done

Tested code with HW

- Successfully measured raw cell voltages from each GPIO pin on the daisy chained dc2350A-B boards
- Check that the functions to get the average, min, and max cell temperature are working

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/BMS/Src/Io/Io_CellTemperatures.c [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    size_t   current_chip,
+    size_t   current_register_group,
+    uint8_t *rx_raw_thermistor_voltages);
+
+/**
+ * Read raw thermistor voltages from the cell monitoring chips
+ * @return EXIT_CODE_OK if raw thermistor voltages are read successfully from
+ * all cell monitoring chips. Else, EXIT_CODE_ERROR
+ */
+static ExitCode Io_CellTemperatures_ReadRawThermistorVoltages(void);
```
[the remark this answers is not in the document]
Comment (pasitlaothamatas) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:81]: These are static functions so we don't usually place them in the header file

File: boards/BMS/Src/Io/Io_CellTemperatures.c [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+            // [1251.8, 32624.2] ohms.
+            if (thermistor_resistance > temperature_lut[0])
+            {
+                return EXIT_CODE_OUT_OF_RANGE;
+            }
+            if (thermistor_resistance <
+                temperature_lut[SIZE_OF_TEMPERATURE_LUT - 1])
+            {
+                return EXIT_CODE_OUT_OF_RANGE;
+            }
```
[the remark this answers is not in the document]
Comment (pasitlaothamatas) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:225]: good catch thanks

File: boards/BMS/Src/Io/Io_CellTemperatures.c [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            current_cell +
+            current_register_group * NUM_OF_THERMISTORS_PER_REGISTER_GROUP;
+        if (current_register_group == AUX_REGISTER_GROUP_C)
+        {
+            // Subtract curr_column to ignore the reference voltage read back
+            // from aux register group B.
+            curr_column--;
+        }
+        raw_thermistor_voltages[current_chip][curr_column] =
+            (uint16_t)raw_thermistor_voltage;
```
[the remark this answers is not in the document]
Comment (pasitlaothamatas) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:110]: Yeah I remember getting a GCC typecast error which I solved by using a uint32_t, but this seemed to fix it. Nice find.

        (uint16_t)((rx_raw_thermistor_voltages[raw_thermistor_voltages_index]) |
            ((rx_raw_thermistor_voltages[raw_thermistor_voltages_index + 1]
              << 8)));

>

File: boards/BMS/Src/Io/Io_CellTemperatures.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (alxshev) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:81]: Ah ok that's good to know

File: boards/BMS/Src/Io/Io_CellTemperatures.c [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+static ExitCode Io_CellTemperatures_ReadRawThermistorVoltages(void)
+{
+    uint16_t aux_register_group_cmd;
+    uint8_t  tx_cmd[NUM_OF_CMD_BYTES];
+    uint8_t  rx_thermistor_resistances
+        [NUM_OF_RX_BYTES * NUM_OF_CELL_MONITOR_CHIPS] = { 0 };
+
+    // The command used to start auxiliary (GPIO) measurements.
+    const uint16_t ADAX = 0x460 + (MD << 7) + CHST;
```
Comment (alxshev) [inline boards/BMS/Src/Io/Io_CellTemperatures.c:137]: image
This is the command being called right?

Comment (alxshev) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:137]: I'm just wondering why it has CHG rather than CHST

Comment (pasitlaothamatas) [reply boards/BMS/Src/Io/Io_CellTemperatures.c:137]: Great catch. Luckily both values were set to 0

File: boards/BMS/Src/Io/Io_CellTemperatures.c [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    return (received_pec15 == calculated_pec15) ? EXIT_CODE_OK
+                                                : EXIT_CODE_ERROR;
+}
+
+static ExitCode Io_CellTemperatures_ReadRawThermistorVoltages(void)
+{
+    uint16_t aux_register_group_cmd;
+    uint8_t  tx_cmd[NUM_OF_CMD_BYTES];
+    uint8_t  rx_thermistor_resistances
+        [NUM_OF_RX_BYTES * NUM_OF_CELL_MONITOR_CHIPS] = { 0 };
```
Comment (alxshev) [inline boards/BMS/Src/Io/Io_CellTemperatures.c:134]: Why's the first element being set to 0?

**Review by alxshev — APPROVED** [review]

**Pull request merged**